internal/export/ingest_pipelines_test.go

```diff
@@ -0,0 +1,194 @@
+
+	estest "github.com/elastic/elastic-package/internal/elasticsearch/test"
+)
+
+func TestExportIngestPipelines(t *testing.T) {
+	// Files for each suite are recorded automatically on first test run.
+	// To add a new suite:
+	// - Configure it here.
+	// - Install the package in a running stack.
+	// - Configure environment variables for this stack (eval "$(elastic-package stack shellinit)").
+	// - Run tests.
+	// - Check that recorded files make sense and commit them.
+	// To update a suite:
+	// - Reproduce the scenario as described in the comments.
+	// - Remove the files that you want to update.
+	// - Follow the same steps to create a new suite.
+	// - Check if the changes are the expected ones and commit them.
+	suites := []*ingestPipelineExportSuite{
+		// To reproduce the scenario:
+		// - Start the stack with version 8.17.4.
+		// - Install apache package (2.0.0).
+		// - Install dga package (2.3.0).
+		&ingestPipelineExportSuite{
+			PipelineIds: []string{
+				"logs-apache.access-2.0.0",
+				"2.3.0-ml_dga_ingest_pipeline",
+			},
+			ExportDir: "./testdata/elasticsearch-8-export-pipelines",
+			RecordDir: "./testdata/elasticsearch-8-mock-export-pipelines",
+			Matcher:   ingestPipelineRequestMatcher,
+		},
+	}
+
+	for _, s := range suites {
+		suite.Run(t, s)
+	}
+}
+
+type ingestPipelineExportSuite struct {
+	suite.Suite
+
+	PipelineIds []string
+	ExportDir   string
+	RecordDir   string
+	Matcher     cassette.MatcherFunc
+}
+
+func (s *ingestPipelineExportSuite) SetupTest() {
+	_, err := os.Stat(s.ExportDir)
+	if errors.Is(err, os.ErrNotExist) {
+		client, err := stack.NewElasticsearchClient()
+		s.Require().NoError(err)
+
+		writeAssignments := createTestWriteAssignments(s.PipelineIds, s.ExportDir)
+
+		err = IngestPipelines((context.Background()), client.API, writeAssignments)
+
+		s.Require().NoError(err)
+	} else {
+		s.Require().NoError(err)
+	}
+}
+
+func (s *ingestPipelineExportSuite) TestExportPipelines() {
+	client := estest.NewClient(s.T(), s.RecordDir, s.Matcher)
+
+	outputDir := s.T().TempDir()
+	writeAssignments := createTestWriteAssignments(s.PipelineIds, outputDir)
+	err := IngestPipelines(context.Background(), client.API, writeAssignments)
+	s.Require().NoError(err)
+
+	filesExpected := countFiles(s.T(), s.ExportDir)
+	filesExported := countFiles(s.T(), outputDir)
+	s.Assert().Equal(filesExpected, filesExported)
+
+	filesFound := countFiles(s.T(), outputDir)
+	s.Assert().Equal(filesExpected, filesFound)
+
+	assertEqualExports(s.T(), s.ExportDir, outputDir)
+}
+
+func createTestWriteAssignments(pipelineIDs []string, outputDir string) PipelineWriteAssignments {
+	writeAssignments := make(PipelineWriteAssignments)
+
+	for _, pipelineID := range pipelineIDs {
+		writeAssignments[pipelineID] = PipelineWriteLocation{
+			Type:       PipelineWriteLocationTypeRoot,
+			Name:       pipelineID,
+			ParentPath: outputDir,
+		}
+	}
+
+	return writeAssignments
+}
+
+func countFiles(t *testing.T, dir string) (count int) {
```

Comment:
Fair bit of testing utils borrowed from the `dump` package tests. I didn't come across an existing package where shared testing utils might have a home, but since this is the only place this code is repeated, I figured this is fine for now. Otherwise, I could create an `internal/testutils` package if we expect to see more of this type of testing pattern of comparing file output from the CLI.

Reply:
Yep, this is fine by now, but creating a shared package sounds good too, as you prefer.